tests/test_upload.py

```diff
@@ -180,6 +180,46 @@ def test_upload_dataset_to_given_branch(self, accesskey, url, tmp_path):
 
         gas_client.delete_dataset(dataset_name)
 
+    def test_upload_dataset_after_commit(self, accesskey, url, tmp_path):
+        gas_client = GAS(access_key=accesskey, url=url)
+        dataset_name = get_dataset_name()
+        gas_client.create_dataset(dataset_name, is_fusion=True)
+
+        dataset = FusionDataset(name=dataset_name)
+        dataset._catalog = Catalog.loads(CATALOG)
+        dataset.notes.is_continuous = True
+        segment = dataset.create_segment("Segment1")
+        segment.sensors.add(Sensor.loads(LIDAR_DATA))
+
+        path = tmp_path / "sub"
+        path.mkdir()
+        for i in range(10):
+            frame = Frame()
+            local_path = path / f"hello{i}.txt"
+            local_path.write_text("CONTENT")
+            data = Data(local_path=str(local_path))
+            data.label = Label.loads(LABEL)
+            frame[LIDAR_DATA["name"]] = data
+            segment.append(frame)
+
+        dataset_client = gas_client.upload_dataset(dataset)
+        dataset_client.commit("test")
+        dataset1 = FusionDataset(name=dataset_name, gas=gas_client)
```

Comment:
Is possible to use `dataset_remote` and `segment_remote` to distinguish with the local `dataset` and `segment` better?